HW4_Sapozhnikov/prototool.py

```diff
@@ -0,0 +1,564 @@
+        for j in range(1, len_seq2 + 1):
+            if seq_on[i - 1] == seq2[j - 1]:
+                match_score = score_matrix[i - 1][j - 1] + match
+            else:
+                match_score = score_matrix[i - 1][j - 1] + mismatch
+
+            delete_score = score_matrix[i - 1][j] + gap
+            insert_score = score_matrix[i][j - 1] + gap
+
+            # Calculate the maximum score for the current cell
+            score = max(0, match_score, delete_score, insert_score)
+
+            # Update the score matrix and traceback matrix
+            score_matrix[i][j] = score
+
+            if score > alignment_score:
+                alignment_score = score
+                max_i, max_j = i, j
+
+            if score == match_score:
+                traceback_matrix[i][j] = "match"
+            elif score == delete_score:
+                traceback_matrix[i][j] = "delete"
+            elif score == insert_score:
+                traceback_matrix[i][j] = "insert"
+            else:
+                traceback_matrix[i][j] = "none"
+
+    # Traceback to find the aligned sequences
+    aligned_seq_on = []
+    aligned_seq2 = []
+
+    counter_identity: int = 0
+    counter_gaps: int = 0
+
+    i, j = max_i, max_j
+
+    while i > 0 and j > 0:
+        if traceback_matrix[i][j] == "match":
+            aligned_seq_on.append(seq_on[i - 1])
+            aligned_seq2.append(seq2[j - 1])
+            counter_identity += 1
+            i -= 1
+            j -= 1
+        elif traceback_matrix[i][j] == "delete":
+            aligned_seq_on.append(seq_on[i - 1])
+            aligned_seq2.append("-")
+            counter_gaps += 1
+            i -= 1
+        elif traceback_matrix[i][j] == "insert":
+            aligned_seq_on.append("-")
+            aligned_seq2.append(seq2[j - 1])
+            counter_gaps += 1
+            j -= 1
+        else:
+            break
+
+    # Reverse the aligned sequences
+    aligned_seq_on = "".join(aligned_seq_on[::-1])
+    aligned_seq2 = "".join(aligned_seq2[::-1])
+
+    alignment_length = (len(aligned_seq_on)
+                        if len(aligned_seq_on) < len(aligned_seq2)
+                        else len(aligned_seq2))
+
+    # Form an output dictionary
+    alignment_dict['aligned_seq_on'] = aligned_seq_on
+
+    identity = round(counter_identity/alignment_length, 4)
+
+    alignment_dict[f'aligned_seq{seq_id+1}'] = {'seq': aligned_seq2,
+                                                'length': alignment_length,
+                                                'score': alignment_score,
+                                                'identity': identity,
+                                                'gaps': counter_gaps}
+
+    # Prettify an alignment output
+    if prettify is True:
+        prettify_alignment(aligned_seq_on, aligned_seq2)
+    else:
+        pass
+
+    return alignment_dict
+
+
+def count_protein_molecular_weight(*seqs_list: Union[List[str], str]) -> dict:
```

Comment:
```suggestion
def calc_protein_molecular_weight(*seqs_list: Union[List[str], str]) -> dict:
```

Все-таки count это про какие-то счестчики (1, 2, 3, ...). Более того, скорее целочисленные. А тут  мы вычисляем массу (хоть формула и простая, но вычисляем). Поэтому `calculate` или `calc`.